typescript/index.d.ts

```diff
@@ -77,7 +77,7 @@ declare namespace WoT {
     export type ThingDescription = { [key: string]: any; };
 
 
-    export type DataSchemaValue = any;
+    export type DataSchemaValue = (string | number | any[] | object | null);
```

Comment:
I am unsure about the array.

We could change `any[]` to `Array<Type>` where `Type` would be `any` or the list of types again? Not sure what makes more sense.

I am more in favor to use the `plain` array construct vs the class style...

Reply:
what's wrong with `DataSchemaValue[]`? ts cannot handle recursive declarations?

Reply:
https://www.typescriptlang.org/play?#code/C4TwDgpgBAIghsOBlAxgCwgWzgNTgGwFdoBeKACgGdgAnASwDsBzKAHygcMwCMIa3YCZOiy4CxANoBdAQHtuAKwgpgAzvnwBKANwAoXfgiqAbuIgAuQYlQZseIqSgSARBg2znU7UA

it seems to work

Reply:
Good point. I missed the obvious :-(